awx/main/utils/handlers.py

```diff
@@ -316,6 +317,7 @@ def serialize(cls, message):
     def get_handler_class(self, protocol):
         return HANDLER_MAPPING.get(protocol, AWXNullHandler)
 
+    @cachetools.cached(cache=cachetools.TTLCache(maxsize=128, ttl=3))
     def get_handler(self, custom_settings=None, force_create=False):
```

Comment:
We're calling this **constantly**, and it is absolutely destroying us when logging is enabled; this is a compromise to cache this slightly and give memcached a break when event logging is enabled.